### PR TITLE
Provide ApiSettings in ReferenceApiModule, using parsed yaml settings

### PR DESCRIPTION
**Pull request by rtannenbaum**
Unlike ExtensionContext flags, which may be arbitrary, validate the expected API flags baseUrl and baseApiUrl with Jackson.

ApiSettings is injected throughout the Reference Api server

#171  #202

### CODE REVIEW

**Comment by rtannenbaum** [comment]
Let me know if this makes any sense... I don't think we should be using ExtensionContext for this

File: extensions/cloud/portability-cloud-google/config/common.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Google cloud settings. See environments/ for environment-specific settings.
+cloud: GOOGLE
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/config/common.yaml]: Can we add a list of example options to the comment

File: libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.util.Map;
+
+// TODO(rtannenbaum): Move this out of security/ into a new module config/. Couldn't figure it
+// out in Intellij.
+public class ConfigUtils {
+  public static Map<String, Object> parse(InputStream in) throws IOException {
```
Comment (holachuy) [inline libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java]: Document this public method

File: libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    return mapper.readValue(in, Map.class);
+  }
+
+  public static InputStream getSettingsCombinedInputStream(ImmutableList<String> settingsFiles) {
```
Comment (holachuy) [inline libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java]: same

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java [hunk 4]
```diff
@@ -28,13 +29,16 @@
...
 import org.dataportabilityproject.launcher.impl.TypeManagerImpl;
+import org.dataportabilityproject.security.ConfigUtils;
 
 /**
  * {@link ExtensionContext} used by the worker.
  */
 final class WorkerExtensionContext implements ExtensionContext {
-  private static final String CLOUD_SETTINGS_PATH = "config/settings.yaml";
-  private static final String CLOUD_ENV_SETTINGS_PATH = "config/env/settings.yaml";
+  private static final String WORKER_SETTINGS_PATH = "config/worker.yaml";
```
Comment (holachuy) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: Are all these distinct? Can they be externally configured?

**Comment by holachuy** [comment]
I patched this in and tried to run the following command but it didn't find a cloud type. Do we have to update a config file?

./bin/build_and_run_jar.sh worker local
ERROR: Must provide a cloud type, e.g. 'local', 'google', or 'microsoft'

**Comment by rtannenbaum** [comment]
@holachuy that's broken in master as well. I can fix that in my next PR when I make the script compile the yaml files into the jar

File: extensions/cloud/portability-cloud-google/config/common.yaml (hunk 1 above)
Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/config/common.yaml]: I think this is the only common setting left actually now that encryptionFlow and serviceProviderClasses are deprecated

File: libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java (hunk 3 above)
Comment (rtannenbaum) [reply libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java]: done

File: libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java (hunk 2 above)
Comment (rtannenbaum) [reply libraries/security/src/main/java/org/dataportabilityproject/security/ConfigUtils.java]: done

**Comment by seehamrun** [comment]
Sorry that might have been me that broke it - I added a 3rd option in #224 so we can configure which cloud package to compile into the jar

File: bin/build_and_run_jar.sh [hunk 5]
```diff
@@ -56,9 +56,9 @@ gradle=$(which gradle)|| { echo "Gradle (gradle) not found. Please install it an
 
 # Copy settings yaml files from ENV/settings/ into $SRC_DIR/src/main/resources/
 # TODO: this script should take cloud extension name as a param, in addition to ENV.
-# It should copy settings from extension-cloud-dir/config/settings.yaml into
-# config/settings.yaml, and extension-cloud-dir/config/environments/$ENV/settings.yaml into
-# config/env/settings.yaml.
+# It should copy common and api/worker settings from extension-cloud-dir/config/ into config/, and
```
Comment (seehamrun) [inline bin/build_and_run_jar.sh]: nit: gateway/worker

**Pull request merged**